index.js

```diff
@@ -240,13 +240,19 @@ FSWatcher.prototype._awaitWriteFinish = function(path, threshold, callback) {
   (function awaitWriteFinish (prevStat) {
     fs.stat(fullPath, function(err, curStat) {
       if (err) {
-        delete this._pendingWrites[path];
+        // prevent `unlink` race condition
+        this._getWatchedDir(sysPath.dirname(path))
+          .remove(sysPath.basename(path));
+        if (this._pendingWrites[path]) {  
+          this._pendingWrites[path].cancelWait();
```

Comment:
The issue I observed in the original code that was causing tests to fail regularly was that `unlink` events were being emitted without any prior `add` event nor an entry in `_pendingWrites`. This happened because of the slowness of `fs.stat` calls, causing misalignment between `this._watched` tracking and `this._pendingWrites` (the file shows up in the former, but awf suppressed the corresponding `add` event from being emitted, then never tracked it because it was gone by the time of the `stat`).

Hmm, hard to explain.